csi.proto

```diff
@@ -245,31 +245,19 @@ message VolumeInfo {
 
   // Contains identity information for the created volume. This field is
   // REQUIRED. The identity information will be used by the CO in
-  // subsequent calls to refer to the provisioned volume.
-  VolumeHandle handle = 2;
-}
-
-// VolumeHandle objects are generated by Plugin implementations to
-// serve two purposes: first, to efficiently identify a volume known
-// to the Plugin implementation; second, to capture additional, opaque,
-// possibly sensitive information associated with a volume. It SHALL
-// NOT change over the lifetime of the volume and MAY be safely cached
-// by the CO.
-// Since this object will be passed around by the CO, it is RECOMMENDED
-// that each Plugin keeps the information herein as small as possible.
-// The total bytes of a serialized VolumeHandle must be less than 1 MiB.
-message VolumeHandle {
-  // ID is the identity of the provisioned volume specified by the
-  // Plugin. This field is REQUIRED.
-  // This information SHALL NOT be considered sensitive such that, for
-  // example, the CO MAY generate log messages that include this data.
-  string id = 1;
-
-  // Metadata captures additional, possibly sensitive, information about
-  // a volume in the form of key-value pairs. This field is OPTIONAL.
-  // Since this field MAY contain sensitive information, the CO MUST NOT
-  // leak this information to untrusted entities.
-  map<string, string> metadata = 2;
+  // subsequent calls to refer to the provisioned volume. This field
+  // should not exceed 1MiB.
+  string id = 2;
+
+  // Attributes reflect static properties of a volume and MUST be passed
+  // to volume validation and publishing calls.
+  // Attributes SHALL be opaque to a CO. Attributes SHALL NOT be mutable
+  // and SHALL be safe for the CO to cache. Attributes SHOULD NOT
+  // contain sensitive information. Attributes MAY NOT uniquely identify
+  // a volume. A volume uniquely identified by `id` SHALL always report
+  // the same attributes. This field is OPTIONAL and when present MUST
+  // be passed to volume validation and publishing calls.
+  map<string,string> attributes = 3;
```

Comment:
I guess we should probably have a limit for the encoded size of this field. 1MiB, or something else?